webui/react/src/pages/ProjectDetails.tsx

```diff
@@ -1,67 +1,78 @@
+import { checkmarkRenderer, defaultRowClassName, experimentNameRenderer, experimentProgressRenderer,
+  ExperimentRenderer, expermentDurationRenderer, getFullPaginationConfig, relativeTimeRenderer,
+  stateRenderer,
+  userRenderer } from 'components/Table';
 import TableBatch from 'components/TableBatch';
 import TableFilterDropdown from 'components/TableFilterDropdown';
 import TableFilterSearch from 'components/TableFilterSearch';
 import TagList from 'components/TagList';
-import TaskActionDropdown from 'components/TaskActionDropdown';
-import { cancellableRunStates, deletableRunStates, pausableRunStates,
-  terminalRunStates } from 'constants/states';
 import { useStore } from 'contexts/Store';
 import useExperimentTags from 'hooks/useExperimentTags';
 import { useFetchUsers } from 'hooks/useFetch';
+import useModalProjectNoteDelete from 'hooks/useModal/Project/useModalProjectNoteDelete';
 import useModalCustomizeColumns from 'hooks/useModal/useModalCustomizeColumns';
+import useModalExperimentMove, {
+  Settings as MoveExperimentSettings,
+  settingsConfig as moveExperimentSettingsConfig,
+} from 'hooks/useModal/useModalExperimentMove';
 import usePolling from 'hooks/usePolling';
 import useSettings, { UpdateSettings } from 'hooks/useSettings';
 import { paths } from 'routes/utils';
-import {
-  activateExperiment, archiveExperiment, cancelExperiment, deleteExperiment, getExperimentLabels,
-  getExperiments, killExperiment, openOrCreateTensorBoard,
-  patchExperiment, pauseExperiment, unarchiveExperiment,
-} from 'services/api';
+import { activateExperiment, addProjectNote, archiveExperiment, cancelExperiment, deleteExperiment,
+  getExperimentLabels, getProject, getProjectExperiments,
+  killExperiment, openOrCreateTensorBoard, patchExperiment, pauseExperiment,
+  setProjectNotes,
+  unarchiveExperiment } from 'services/api';
 import { Determinedexperimentv1State, V1GetExperimentsRequestSortBy } from 'services/api-ts-sdk';
 import { encodeExperimentState } from 'services/decoder';
 import Icon from 'shared/components/Icon/Icon';
+import Message, { MessageType } from 'shared/components/Message';
+import Spinner from 'shared/components/Spinner';
 import { isEqual } from 'shared/utils/data';
+import { isNotFound } from 'shared/utils/service';
 import {
-  ExperimentAction as Action, CommandTask, ExperimentItem, RunState,
+  ExperimentAction as Action,
+  CommandTask,
+  ExperimentItem,
+  Note,
+  Project,
+  ProjectExperiment,
+  RunState,
 } from 'types';
 import handleError from 'utils/error';
+import {
+  getActionsForExperimentsUnion,
+  getProjectExperimentForExperimentItem,
+} from 'utils/experiment';
 import { alphaNumericSorter } from 'utils/sort';
-import { isTaskKillable, taskFromExperiment } from 'utils/task';
 import { getDisplayName } from 'utils/user';
 import { openCommand } from 'wait';
 
 import { RecordKey } from '../shared/types';
 import { ErrorLevel } from '../shared/utils/error';
```

Comment:
**nit:** don't have to, but convenient to just have `shared/...` instead of `../shared/...`